### PR TITLE
Add Zitadel provider

### PR DESCRIPTION
**Pull request by tbhaxor**

### CODE REVIEW

File: src/Zitadel/composer.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        }
+    },
+    "authors": [
+        {
+            "name": "Gurkirat Singh",
+            "email": "tbhaxor@gmail.com"
+        }
+    ],
+    "require": {
+        "php": "^8.0",
```
Comment (atymic) [inline src/Zitadel/composer.json:19]:
```suggestion
        "php": "^8.1",
```

File: tools/docgen.php [hunk 2]
```diff
@@ -62,10 +62,11 @@
 
 DOC;
 
-$directories = array_map('basename', glob('../src'.'/*', GLOB_ONLYDIR));
+$providersDir = realpath(__DIR__ . '/../src');
+$directories = array_map('basename', glob($providersDir . '/*', GLOB_ONLYDIR));
 
 foreach ($directories as $provider) {
-    $path = sprintf('%s/../src/%s/README.md', __DIR__, $provider);
```
Comment (atymic) [inline tools/docgen.php:68]: Was there a reason for this change?

Comment (tbhaxor) [reply tools/docgen.php:68]: Yes, which this you can run the docgen file from any directory. For example,

```console
php ../../tools/docgen.php
```

**Comment by tbhaxor** [comment]
@atymic The provider is not available on packagist.com and also the readme contents are not available on the website https://socialiteproviders.com/Zitadel/.

**Comment by atymic** [comment]
You'll need to PR to the website for the second one, see email for first.

**Pull request merged**